language-support/java/codegen/src/test/scala/com/digitalasset/daml/lf/codegen/CodeGenRunnerTests.scala

```diff
@@ -58,14 +58,53 @@ final class CodeGenRunnerTests extends AnyFlatSpec with Matchers {
 
   it should "read interfaces from a single DAR file with a prefix" in {
 
-    val scope = CodeGenRunner.configureCodeGenScope(Map(testDar -> Some("PREFIX")), Map.empty)
+    val scope = CodeGenRunner.configureCodeGenScope(Map(testDar -> Some("prefix")), Map.empty)
 
     assert(scope.signatures.map(_.packageId).length === dar.all.length)
     assert(scope.packagePrefixes.size === dar.all.length)
-    assert(scope.packagePrefixes.values.forall(_ === "PREFIX"))
+    assert(scope.packagePrefixes.values.forall(_ === "prefix"))
     assert(scope.toBeGenerated === Set.empty)
   }
 
+  it should "read interfaces from 2 DAR files with same content and same prefixes" in {
+
+    val scope =
+      CodeGenRunner.configureCodeGenScope(
+        Map(testDar -> Some("prefix"), testDarWithSameSrcAndProjectNamePathDar -> Some("prefix")),
+        Map.empty,
+      )
+
+    assert(scope.signatures.length === dar.all.length)
+    assert(scope.packagePrefixes.size === dar.all.length)
+    assert(scope.packagePrefixes.values.forall(_ === "prefix"))
+    assert(scope.toBeGenerated === Set.empty)
+  }
+
+  it should "fail if read interfaces from 2 DAR files with same content but different prefixes" in {
+    assertThrows[IllegalArgumentException] {
+      CodeGenRunner.configureCodeGenScope(
+        Map(testDar -> Some("prefix1"), testDarWithSameSrcAndProjectNamePathDar -> Some("prefix2")),
+        Map.empty,
+      )
+    }
+  }
+
+  it should "read interfaces from 2 DAR files with one is depending on other packages using data_dependencies" in {
```

Comment:
This test case covers the example here:
https://github.com/digital-asset/daml/issues/14976#issuecomment-1315826638